website/content/docs/plugins/install.mdx

```diff
@@ -97,7 +97,7 @@ Refer to [`packer init` command](/packer/docs/commands/init) for additional info
 
 ## Install a plugin under development
 
-You can install and use plugins that are still under development and report as `dev` versions. But if a non-development version of the plugin binary is available, Packer installs the non-development version according to the version constraints configured in the Packer template and uses it in the build.
+You can install and use plugins that are still under development and report as `dev` versions. But if a final version of the plugin binary is available, Packer installs the final version according to the version constraints specified in the Packer template and uses it in the build.
```

Comment:
I would still suggest a reword here, if only to avoid confusion regarding packer init/packer plugins install, development binaries can only be installed with --path, remote installation only supports final releases, so regardless of the specified constraints in the template, those commands in non-local mode will never install a pre-release version of a plugin (or at least if it does, that's a bug)

Reply:
OK, just to be clear, to install a `-dev` plugin, you have to run `packer plugins install --path`. You cannot specify a `dev` plugin in the `required_plugins` field and run `packer init`.  Is that correct? If so, I'm not sure I understand the value of the rest of this section.

Reply:
Yes exactly, `-dev` constraints are not supported (not sure if they're explicitly rejected though, I'll have to make sure of that), and the only supported method to install dev versions is through `packer plugins install --path`.

Reply:
https://github.com/hashicorp/packer/pull/12999
^ So it wasn't supported, but the error message was a bit lacklustre, I've opened a PR to make it clearer that pre-releases are not supported as part of constraints.